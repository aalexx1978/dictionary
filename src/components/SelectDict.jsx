import React from "react";
import { FormGroup, ControlLabel, FormControl } from "react-bootstrap";

class SelectDict extends React.Component {
    constructor(props) {
        super(props);
        this.handleChange = this.handleChange.bind(this);
    }
    handleChange(e) {
        let selected = parseInt(e.target.value, 10);
        this.props.appActions.setDictionary(selected);
    }
    render() {
        const { dictionaries, dictionary } = this.props;
        return (
            <FormGroup controlId="formControlsSelect">
                <ControlLabel>Select Dictionary to apply</ControlLabel>
                <FormControl
                    defaultValue={dictionary}
                    onChange={e => this.handleChange(e)}
                    componentClass="select"
                    placeholder="select"
                >
                    <option value={0}>Original</option>
                    {dictionaries.map(item => {
                        return (
                            <option key={item.id} value={item.id}>
                                {item.name}
                            </option>
                        );
                    })}
                </FormControl>
            </FormGroup>
        );
    }
}
export default SelectDict;
