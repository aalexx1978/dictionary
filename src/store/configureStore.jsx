import { createStore, applyMiddleware, compose } from "redux";
import rootReducer from "../reducers/allReducers";
import thunk from "redux-thunk";
import { logger } from "redux-logger";
const store = createStore(
    rootReducer,
    window.__REDUX_DEVTOOLS_EXTENSION__ &&
        window.__REDUX_DEVTOOLS_EXTENSION__(),
    compose(applyMiddleware(thunk, logger))
);

export default function configureStore() {
    return store;
}
