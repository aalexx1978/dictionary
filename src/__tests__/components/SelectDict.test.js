import React from "react";
import Enzyme from "enzyme";
import { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import toJson from "enzyme-to-json";
Enzyme.configure({ adapter: new Adapter() });
import configureStore from "redux-mock-store";
const dict = [
    {
        name: "Dictionary 3",
        Stonegrey: "Dark",
        "Midnight Black": "Black Dark",
        "Mystic Silver": "Silver Special",
        id: 3
    }
];
const sel = 0;

import SelectDict from "../../components/SelectDict";
describe("<SelectDict />", () => {
    describe("render()", () => {
        test("renders the component", () => {
            const wrapper = shallow(
                <SelectDict dictionaries={dict} dictionary={sel} />
            );
            const component = wrapper.dive();

            expect(toJson(component)).toMatchSnapshot();
        });
    });
});
