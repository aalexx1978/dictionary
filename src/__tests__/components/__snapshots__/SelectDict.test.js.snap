// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectDict /> render() renders the component 1`] = `
<div
  className="form-group"
>
  <ControlLabel
    bsClass="control-label"
    srOnly={false}
  >
    Select Dictionary to apply
  </ControlLabel>
  <FormControl
    bsClass="form-control"
    componentClass="select"
    defaultValue={0}
    onChange={[Function]}
    placeholder="select"
  >
    <option
      value={0}
    >
      Original
    </option>
    <option
      key="3"
      value={3}
    >
      Dictionary 3
    </option>
  </FormControl>
</div>
`;
